metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/PoolStatsInterceptor.java

```diff
@@ -57,10 +61,13 @@ public void disconnected(final ConnectionPool parent, final PooledConnection con
     }
 
     private void publishMetric(final ConnectionPool parent) {
-        if (parent != null && metricNameTotal != null) {
-            DynamicGauge.set(metricNameTotal, parent.getSize());
-            DynamicGauge.set(metricNameActive, parent.getActive());
-            DynamicGauge.set(metricNameIdle, parent.getIdle());
+        if (parent != null
+            && metricNameTotalGauage != null
```

Comment:
They can't be instantiated in the constructor, as the "metricName" are unknown at that time.